frontend/app/common/fetcher.ts

```diff
@@ -13,38 +14,60 @@ export const XSRF_HEADER = 'X-XSRF-TOKEN';
 /** Cookie field with XSRF token */
 export const XSRF_COOKIE = 'XSRF-TOKEN';
 
-type Methods = typeof METHODS;
-type FetchInit = Omit<RequestInit, 'headers'> & {
-  headers?: Record<string, string>;
-  json?: unknown;
-  query?: Record<string, string | number | undefined>;
-};
-type FetcherObject = Record<Methods[number], <T>(url: string, params?: FetchInit) => Promise<T>>;
+type QueryParams = Record<string, string | number | undefined>;
+type Payload = BodyInit | Record<string, unknown> | null;
+type FetcherMethods = Record<'get' | 'delete', <T>(url: string, query?: QueryParams) => Promise<T>> &
+  Record<'put' | 'post', <T>(url: string, query?: QueryParams, body?: Payload) => Promise<T>>;
+
+export function stringifyUrl(uri: string, query: QueryParams) {
+  const queryEntries = Object.entries(query);
+  const siteIdParam = `site=${encodeURIComponent(siteId)}`;
+  const baseUrl = uri.indexOf('/auth') === 0 ? BASE_URL : `${BASE_URL}${API_BASE}`;
```

Comment:
I propose to extract that "either BASE_URL or API_BASE as well" to some higher level, so the base URL would be already clear by the time you call that function.

`stringifyUrl` currently does more than the name implies and also it should know some internals about the URL which is passed to it, that logic should be moved up the stack by wrapping this functionality into separate function or object, like `fetcher` vs `authFetcher`, or `fetcher` vs `apiFetcher` which would be distinct only by different baseURL, which would be determined by the time of object creation and not per URL passed to it.

Reply:
Yep. You are right. I thought about making some sort of fabric and pass different base URLs.